src/Controller/CRUDController.php

```diff
@@ -314,10 +314,6 @@ public function editAction($id = null)
         $form->handleRequest($request);
 
         if ($form->isSubmitted()) {
-            //TODO: remove this check for 4.0
-            if (method_exists($this->admin, 'preValidate')) {
```

Comment:
Isn't it a BC if you remove this?

Reply:
Technically yes, but it didn't work as expected. So this is a bugfix.

Reply:
`preValidate` method is still call but at the right moment now. So i think that we can say it's BC compatible.

Reply:
You need to keep the if, wherever you call the method because it could not be there.